core/src/main/java/org/commonjava/maven/ext/core/state/RESTState.java

```diff
@@ -54,7 +62,16 @@ public void initialise( Properties userProps )
                                                                    String.valueOf( DefaultTranslator.CHUNK_SPLIT_COUNT ) ) );
         restSuffixAlign = Boolean.parseBoolean( userProps.getProperty( "restSuffixAlign", "true" ) );
 
-        restEndpoint = new DefaultTranslator( restURL, restMaxSize, restMinSize, repositoryGroup, vState.getIncrementalSerialSuffix() );
+        String restHeadersProperty = userProps.getProperty( "restHeaders", "" );
+        if ( !StringUtils.isEmpty( restHeadersProperty ) )
+        {
+            restHeaders = Arrays.stream( restHeadersProperty.split( "," ) )
+                                .map( Header::new )
+                                .collect( Collectors.toMap( Header::getName, Header::getValue,
```

Comment:
Isn't using Header a bit 'overkill' - why not split on comma and split on colon ?

Reply:
I agree. I thought it might be easier to handle the edge cases like a null or empty header name, but I've converted it to not use a class.